### PR TITLE
SPIN-2076 allow custom message on stage / pipeline notifications

### PR DESCRIPTION
**Pull request by robfletcher**
This change enables us to specify a custom message that gets appended to Slack or Email notifications. Echo changes in place already.

### CODE REVIEW

**Comment by robfletcher** [comment]
@anotherchrisberry @zanthrash @icfantv PTAL

File: app/scripts/modules/core/notification/modal/editNotification.controller.modal.js [hunk 1]
```diff
@@ -63,6 +63,10 @@ module.exports = angular
       $uibModalInstance.close(vm.notification);
     };
 
+    vm.supportsCustomMessage = function(notification) {
```
Comment (robfletcher) [inline app/scripts/modules/core/notification/modal/editNotification.controller.modal.js]: I don't believe so — the method is invoked based on the state of the notification type select box which has no empty option.

Comment (icfantv) [inline app/scripts/modules/core/notification/modal/editNotification.controller.modal.js]: ...i don't know that it matters...

File: app/scripts/modules/core/notification/modal/editNotification.html [hunk 2]
```diff
@@ -10,17 +10,25 @@ <h3 data-purpose="modal-header">Edit Notification</h3>
...
-            <input
-              type="checkbox"
-              name="when"
-              value="{{wo}}"
-              ng-model="selectedWhenOptions[wo]"
-              ng-checked="editNotification.notification.when.includes(wo)">
-            {{wo|notificationWhen:level}}
-
-          </label>
+        <div class="col-sm-9" ng-class="$first ? '' : 'col-sm-offset-3'" ng-repeat="wo in editNotification.whenOptions">
```
Comment (icfantv) [inline app/scripts/modules/core/notification/modal/editNotification.html]: would you mind please renaming `wo` to `whenOption`?

**Comment by icfantv** [comment]
just the one nit from me.

**Comment by anotherchrisberry** [comment]
LGTM. It would be nice to rewrite it to TS, but that is asking quite a bit, and this is a pretty minor change.

**Pull request merged**